scripts/ci

```diff
@@ -0,0 +1,75 @@
+#!/usr/bin/env bash
+
+platform=`uname`
+echo platform=${platform}
+echo "running against commit: "`git log --oneline --no-decorate -n 1`
+
+set -x
+set -e
+
+function format_check() {
+    scripts/format
+    git diff --no-ext-diff | head -n 20 > format_errors
+    num_lines=`cat format_errors | wc -l`
+    if [ "0" != "${num_lines}" ] ; then
+        echo CODE FORMATTING ERRORS:
+        cat format_errors
+        exit 1
+    fi
+}
+
+function build() {
+
+    dir_name=$1
+    build_type=$2
+
+    # Enable warnings-as-errors
+
+    cxx_flags="-Werror"
+
+    if [ "Darwin" == "${platform}" ] ; then
+        export PATH="/usr/local/opt/llvm/bin:/usr/local/bin:${PATH}"
+        export PKG_CONFIG_PATH="/usr/local/opt/openssl/lib/pkgconfig"
+        export LIBRARY_PATH="/usr/local/opt/openssl/lib"
+        export LDFLAGS="-L/usr/local/opt/llvm/lib -L-L/usr/local/lib"
+        export CPPFLAGS="-I/usr/local/opt/llvm/include -I/usr/local/include"
+
+        cxx_flags="${cxx_flags} -Wno-deprecated-declarations"
+    fi
+
+    cmake_flags="-DFAST_TESTS_ONLY=ON -DCMAKE_BUILD_TYPE=${build_type}"
```

Comment:
Why skip the prover tests? It's a slow one but this is the one that tests the entire circuit.

Reply:
We want to make sure all tests pass before merging

Reply:
I've enabled the prover_test as well.
I originally disabled it for these per-commit builds to get faster turnaround time on results, and wasn't sure if travis had some timeout.  But let's see how it goes.  Definitely better if we can run it on every check-in.

Reply:
prover_test is enabled and running now (builds take approx 22 mins).

Reply:
Yes, let's see how it goes. Thanks!